tensorboard/plugins/profile/tf_profile_dashboard/tf-profile-dashboard.html

```diff
@@ -151,15 +152,22 @@ <h3>No profile data was found.</h3>
         value: 0,
         observer: '_selectedDatasetChanged'
       },
-      _currentTool: {
-        type: String,
-        computed: '_getCurrentToolName(selectedTool, _activeTools)',
-        notify: true,
-      },
       selectedTool: {
         type: Number,
         notify: true,
         value: 0,
+        observer: '_selectedToolChanged'
+      },
+      // This is combination of selectedDataset and selectedTool.
+      // Most significant bits contains index of datasets;
+      // Lest Significant bits contains index of tools;
+      // With this arrangement, we can achieve cascading Polymer effects
+      // propagation and atomic change of both dataset and tool.(e.g.
+      // when dataset change, the tool set to the first available one).
+      currentSelected: {
```

Comment:
I discussed with the team, and we prefer to avoid bit manipulation within JavaScript. Could we replace this number containing bits with a JS object possessing 2 properties: `datasetIndex` and `toolIndex`? And then change other logic appropriately?